### PR TITLE
Added definition to not open pairing window after boot.

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
#### Problem
Currently pairing window and in result BLE advertisement is started automatically after application start, what is not compatible with the spec and there is no way to configure it.

 #### Summary of Changes
* Added definition CHIP_DEVICE_CONFIG_ENABLE_PAIRING_AUTOSTART, set to 1 by default.
* Added condition checking definition value before opening pairing window in the InitServer()
* Added Kconfig option for nrfconnect to allow easy disable pairing autostart
* Aligned nrfconnect samples to open window pairing in the button handlers instead of enabling BLE advertising

### CODE REVIEW

File: src/app/server/Server.cpp [hunk 1]
```diff
@@ -282,7 +282,9 @@ void InitServer(AppDelegate * delegate)
     }
     else
     {
+#if CHIP_DEVICE_CONFIG_ENABLE_PAIRING_AUTOSTART
```
Comment (LuDuda) [inline src/app/server/Server.cpp:285]: nit: move #if before else statement.

```
#if ...
else
{
...
}
#endif
```

**Review by LuDuda — APPROVED** [review]

File: src/app/server/Server.cpp (hunk 1 above)
Comment (kkasperczyk-no) [reply src/app/server/Server.cpp:285]: Having if statement without else is not MISRA compliant and I noted that in community empty else blocks are started to appear

File: examples/lighting-app/nrfconnect/main/AppTask.cpp [hunk 2]
```diff
@@ -381,30 +381,37 @@ void AppTask::StartBLEAdvertisementHandler(AppEvent * aEvent)
     if (aEvent->ButtonEvent.PinNo != BLE_ADVERTISEMENT_START_BUTTON)
         return;
 
-    if (!sNFC.IsTagEmulationStarted())
+    if (!chip::DeviceLayer::ConnectivityMgr().IsThreadProvisioned())
```
Comment (Damian-Nordic) [inline examples/lighting-app/nrfconnect/main/AppTask.cpp:384]: nit:
```
if (chip::DeviceLayer::ConnectivityMgr().IsThreadProvisioned())
  return;
```
would reduce the indentation level.

File: examples/lighting-app/nrfconnect/main/AppTask.cpp [hunk 3]
```diff
@@ -381,30 +381,37 @@ void AppTask::StartBLEAdvertisementHandler(AppEvent * aEvent)
...
-        LOG_INF("NFC Tag emulation is already started");
-    }
 
-    if (!ConnectivityMgr().IsBLEAdvertisingEnabled())
-    {
-        ConnectivityMgr().SetBLEAdvertisingEnabled(true);
-        LOG_INF("Enabled BLE Advertisement");
+        if (!ConnectivityMgr().IsBLEAdvertisingEnabled())
+        {
+            OpenDefaultPairingWindow();
```
Comment (Damian-Nordic) [inline examples/lighting-app/nrfconnect/main/AppTask.cpp:404]: should we check an error here?

**Review by Damian-Nordic — APPROVED** [review]

**Review by Damian-Nordic — APPROVED** [review]

File: src/app/server/Server.cpp (hunk 1 above)
Comment (LuDuda) [reply src/app/server/Server.cpp:285]: Ok we can leave it, though as you mentioned the code is full of such code.

**Review by LuDuda — APPROVED** [review]

**Pull request closed without merging**